extensions/standard-processors/processors/TailFile.cpp

```diff
@@ -403,102 +403,141 @@ void TailFile::onTrigger(const std::shared_ptr<core::ProcessContext> &context, c
     state_file_ = st_file + "." + getUUIDStr();
   }
   if (!this->state_recovered_) {
-    state_recovered_ = true;
-    // recover the state if we have not done so
     this->recoverState(context);
+    state_recovered_ = true;
+  }
+
+  if (tail_mode_ == Mode::MULTIPLE) {
+    checkForRemovedFiles();
+    checkForNewFiles();
   }
 
-  /**
-   * iterate over file states. may modify them
-   */
+  bool did_something = false;
+
+  // iterate over file states. may modify them
   for (auto &state : tail_states_) {
-    auto fileLocation = state.second.path_;
-
-    checkRollOver(context, state.second, state.first);
-    std::string fullPath = fileLocation + utils::file::FileUtils::get_separator() + state.second.current_file_name_;
-    struct stat statbuf;
-
-    logger_->log_debug("Tailing file %s from %llu", fullPath, state.second.currentTailFilePosition_);
-    if (stat(fullPath.c_str(), &statbuf) == 0) {
-      if ((uint64_t) statbuf.st_size <= state.second.currentTailFilePosition_) {
-        logger_->log_trace("Current pos: %llu", state.second.currentTailFilePosition_);
-        logger_->log_trace("%s", "there are no new input for the current tail file");
-        context->yield();
-        return;
+    did_something |= processFile(context, session, state.first, state.second);
+  }
+
+  if (!did_something) {
+    yield();
+  }
+}
+
+  bool TailFile::processFile(const std::shared_ptr<core::ProcessContext> &context,
+                             const std::shared_ptr<core::ProcessSession> &session,
+                             const std::string &fileName,
+                             TailState &state) {
+    std::string full_file_name = state.fileNameWithPath();
+
+    bool did_something = false;
+
+    if (utils::file::FileUtils::file_size(full_file_name) < state.position_) {
+      std::vector<TailState> rotated_file_states = findRotatedFiles(state);
+      for (TailState &file_state : rotated_file_states) {
+        did_something |= processSingleFile(context, session, file_state.file_name_, file_state);
```

Comment:
std::any_of here as well

Reply:
I have removed all this `did_something` ugliness, and replaced it with `ProcessSession::existsFlowFileInRelationship(Success)`.